crates/uv-workspace/src/workspace.rs

```diff
@@ -1430,6 +1430,34 @@ impl VirtualProject {
         path: &Path,
         options: &DiscoveryOptions,
         cache: &WorkspaceCache,
+    ) -> Result<Self, WorkspaceError> {
+        Self::discover_impl(path, options, cache, false).await
+    }
+
+    /// Equivalent to [`VirtualProject::discover`] but consider it acceptable for
+    /// both `[project]` and `[tool.uv.workspace]` to be missing.
+    ///
+    /// If they are, we act as if an empty `[tool.uv.workspace]` was found.
```

Comment:
Shouldn't we rather forego workspace discovery then and have some kind of `Workspace::for_non_project_pyproject_toml`?

Reply:
So the issue is that we *do* want to do workspace discovery *if we can*. Because we want to respect tool.uv sources/indexes (potentially defined in the parent workspace toml) if they are defined, but we also don't want to error if that kind of stuff is missing.

The entire impl is trying to perform graceful degradation.